common/autoware_trajectory/include/autoware/trajectory/interpolator/detail/interpolator_common_interface.hpp

```diff
@@ -120,8 +120,9 @@ class InterpolatorCommonInterface
       return static_cast<int32_t>(bases_.size()) - 2;
     }
     auto comp = [](const double & a, const double & b) { return a <= b; };
-    return std::distance(bases_.begin(), std::lower_bound(bases_.begin(), bases_.end(), s, comp)) -
-           1;
+    const int32_t idx =
+      std::distance(bases_.begin(), std::lower_bound(bases_.begin(), bases_.end(), s, comp)) - 1;
```

Comment:
If I'm understanding @soblin san's proposal:
```suggestion
    const auto distance = std::distance(bases_.begin(), std::lower_bound(bases_.begin(), bases_.end(), s, comp));
    const int32_t idx = (distance == 0) ? 0 : static_cast<int32_t>(distance) - 1;
```